precompile/config/config.go

```diff
@@ -2,7 +2,7 @@
 // See the file LICENSE for licensing terms.
 
 // Defines the stateless interface for unmarshalling an arbitrary config of a precompile
-package config
+package precompileConfig
```

Comment:
I think if we want to change the package name, we should also change the folder name

Reply:
I'm not %100 sure about Go best practices but I feel like package name should be `precompileconfig`

Reply:
Agree it should be all lowercase. Generally, I think multi-word package names are discouraged. I'd prefer to leave as config and import under the alias `precompileConfig`, but maybe that's too much manual work.

Reply:
I tried that before but I think the lowercase option is better (not a fan of aliasing), feel free to have a look at the previous commit and lmk what you think @aaronbuchwald .

This package is going to be a bit hard to fit with the standard rules.